### PR TITLE
Added new chapter/assembly on Foreman Webhooks

### PR DESCRIPTION
**Pull request by ianf77**
Created files. Included comments from previous PRs.
Rearranged the order of files.
Added more details for shellhooks

[doc] Foreman Webhooks

https://issues.redhat.com/browse/SAT-3106

Cherry-pick into:

* [X] Foreman 3.0
* [X] Foreman 2.5 (Satellite 6.10)
* [ ] Foreman 2.4
* [ ] Foreman 2.3 (Satellite 6.9)

### CODE REVIEW

File: guides/common/modules/con_shellhooks-plugin.adoc [hunk 1]
```diff
@@ -0,0 +1,20 @@
+[id="shellhooks_{context}"]
+= Shellhooks
+
+With webhooks one {Project} event can only be mapped to one API call.
+For advanced integrations, where a single shell script can contain multiple commands, you can install a {SmartProxy} shellhooks plugin that exposes executables using a REST HTTPS API.
+
+A webhook can then be configured to reach out to a {SmartProxy} API to run a predefined shellhook, which can contain commands or can edit files.
```
Comment (ares) [inline guides/common/modules/con_shellhooks-plugin.adoc:7]: which can `e.g.` contain commands or can edit files.

These are just two examples, but of course the shell script can do whatever shell can do.

File: guides/common/modules/con_shellhooks-plugin.adoc [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+Scripts must be placed in `/var/lib/foreman-proxy/shellhooks` as executables with only alphanumeric characters and underscores in the name.
+
+The HTTPS payload is passed using standard input, optional command line arguments can be provided using X-Shellhook-Arg-1 _to_ N.
+
+The HTTP method must be POST.
+An example URL would be: `\https://{smartproxy-example-com}:{smartproxy_port}/shellhooks/my_script`.
+
+Select the *Proxy Authorization* check box to enable each webhook to authorize a call.
```
[the remark this answers is not in the document]
Comment (ianf77) [reply guides/common/modules/con_shellhooks-plugin.adoc:16]: Done

File: guides/common/modules/con_shellhooks-plugin.adoc (hunk 1 above)
Comment (ianf77) [reply guides/common/modules/con_shellhooks-plugin.adoc:7]: Done.

**Comment by lzap** [comment]
Damn, I forgot to cherry pick this one! Sorry, its now in 2.5 and 3.0:

Thank you for your contribution, commits a0b8418a223989ea6a49434e46c1dba834d were cherry-picked into the '3.0' branch.

Thank you for your contribution, commits a0b8418a223989ea6a49434e46c1dba834d were cherry-picked into the '2.5' branch.

**Comment by ianf77** [comment]
Thank you!

**Pull request merged**